### PR TITLE
Register DNS for Ingress/Services Using ALIAS type

### PR DESCRIPTION
**Pull request by goruha**
## what
* Added CNAME support
* Added Ingress support
* Use `ALIAS` for ELBs

## why
* To support Lets Encrypt requirement of A records for all certificates

## who
@cloudposse/engineering

### CODE REVIEW

File: service_listener.go [hunk 1]
```diff
@@ -301,7 +405,7 @@ func updateDNS(r53Api *route53.Route53, hn, hzID, domain, zoneID string) error {
 		HostedZoneId: &zoneID,
 	}
 	if dryRun {
-		glog.Infof("DRY RUN: We normally would have updated %s to point to %s (%s)", zoneID, hzID, hn)
+		glog.Infof("DRY RUN: We normally would have updated %s to point to %s (%s)", zoneID)
```
Comment (osterman) [inline service_listener.go]: why are there 3 `%s` macros, but one value?

File: service_listener.go [hunk 2]
```diff
@@ -115,58 +125,160 @@ func main() {
...
+		glog.Fatalf("Failed to parse selector %v: %v", selector, err)
+		return nil
+	}
+	serviceListOptions := api.ListOptions{
+		LabelSelector: l,
+	}
+
+	services, err := c.Services(api.NamespaceAll).List(serviceListOptions)
+	if err != nil || len(services.Items) == 0 {
+		glog.Fatalf("Failed to list services that used for ingress: %v", err)
```
Comment (osterman) [inline service_listener.go]: "Failed to list services that use ingress: %v"

File: service_listener.go [hunk 3]
```diff
@@ -115,58 +125,160 @@ func main() {
...
+	return dnsRecordType == A || dnsRecordType == CNAME
+}
+
+func getIngressService(c *client.Client)*api.Service {
+	selector := os.Getenv("INGRESS_SERVICE_SELECTOR")
+	if selector == "" {
+		selector = "ingress=endpoint"
+	}
+
+	glog.Infof("User selector %v to find service for ingress", selector)
```
Comment (osterman) [inline service_listener.go]: "Using selector %v to find services for ingress"

File: service_listener.go [hunk 4]
```diff
@@ -311,3 +415,29 @@ func updateDNS(r53Api *route53.Route53, hn, hzID, domain, zoneID string) error {
...
+func makeCNAMETypeRecordSet(hn, hzID, domain string) route53.ResourceRecordSet {
+	return route53.ResourceRecordSet{
+		ResourceRecords: []*route53.ResourceRecord{
+			&route53.ResourceRecord{
+				Value: &hn,
+			},
+		},
+		Name:  &domain,
+		Type:  aws.String("CNAME"),
+		TTL:   aws.Int64(300),
```
Comment (osterman) [inline service_listener.go]: TTL should work like `dnsRecordType` with default value of 300

File: service_listener.go (hunk 1 above)
Comment (goruha) [reply service_listener.go]: Artifact of refactoring...
Done.

File: service_listener.go (hunk 2 above)
Comment (goruha) [reply service_listener.go]: Done

File: service_listener.go (hunk 3 above)
Comment (goruha) [reply service_listener.go]: Done

File: service_listener.go (hunk 4 above)
Comment (goruha) [reply service_listener.go]: Done

**Pull request merged**